modules/mlops/kubeflow-platform/deployspec.yaml

```diff
@@ -0,0 +1,90 @@
+deploy:
+  phases:
+    install:
+      commands:
+      - bash install_build.sh
+    build:
+      commands:
+      # Export all env params specific to the deployment stuff
+      - export ROOT_DIR=$(pwd)
+      - export CLUSTER_NAME=${ADDF_PARAMETER_EKS_CLUSTER_NAME}
+      - export CLUSTER_REGION=${AWS_DEFAULT_REGION}
+      - export INSTALLATION_OPTION=${ADDF_PARAMETER_INSTALLATION_OPTION} #helm
+      - export DEPLOYMENT_OPTION=${ADDF_PARAMETER_DEPLOYMENT_OPTION} #vanilla
+      - export KUBEFLOW_RELEASE_VERSION=${ADDF_PARAMETER_KUBEFLOW_RELEASE_VERSION} #v1.6.1
+      - export AWS_KUBEFLOW_BUILD=${ADDF_PARAMETER_AWS_KUBEFLOW_BUILD} #1.0.0
+      - export AWS_RELEASE_VERSION=${KUBEFLOW_RELEASE_VERSION}-aws-b${AWS_KUBEFLOW_BUILD}
+      ## Create and attach the policy to allow the KF admin role to create/delete roles
+      - export KF_POLICY_NAME=addf-${ADDF_DEPLOYMENT_NAME}-${ADDF_MODULE_NAME}-${AWS_DEFAULT_REGION}-kf
+      - python manage_admin_user.py create ${KF_POLICY_NAME} ${ADDF_PARAMETER_EKS_CLUSTER_MASTER_ROLE_ARN} 
+      # Clone the AWS-on-kubeflow and change to the branch
+      - git clone https://github.com/awslabs/kubeflow-manifests.git && cd kubeflow-manifests
+      - git checkout $AWS_RELEASE_VERSION
+      - git clone --branch ${KUBEFLOW_RELEASE_VERSION} https://github.com/kubeflow/manifests.git upstream
+      - git status
+      - python3.8 -m pip install -r tests/e2e/requirements.txt
+      # Assuming EKS Role
+      - cd $ROOT_DIR/kubeflow-manifests
+      - eval $(aws sts assume-role --role-arn ${ADDF_PARAMETER_EKS_CLUSTER_MASTER_ROLE_ARN} --role-session-name aws-auth-ops | jq -r '.Credentials | "export AWS_ACCESS_KEY_ID=\(.AccessKeyId)\nexport AWS_SECRET_ACCESS_KEY=\(.SecretAccessKey)\nexport AWS_SESSION_TOKEN=\(.SessionToken)\n"')
+      - aws eks update-kubeconfig --name ${CLUSTER_NAME}
+      - kubectl get pods -n kube-system
+      # Deploy Kubeflow here
+      - make deploy-kubeflow INSTALLATION_OPTION=$INSTALLATION_OPTION DEPLOYMENT_OPTION=$DEPLOYMENT_OPTION
+      # Make sure the IRSA for profiles is created with proper annotations, if not, DELETE and redeploy to force it (supports redeployments)
+      - cd $ROOT_DIR/
+      - bash install_role_irsa.sh
+      # Install the NVIDIA Plugin
+      - helm repo add nvdp https://nvidia.github.io/k8s-device-plugin
+      - helm repo update
+      - helm upgrade -i nvdp nvdp/nvidia-device-plugin --version=0.13.0 --namespace nvidia-device-plugin --create-namespace --set-file config.map.config=gpu/nvidia-plugin-configmap.yaml || true
```

Comment:
The version of the nvidia-device-plugin is hard-coded.
It could be nicer to define it as an environment variable?

Reply:
I am torn about this one.  I will evaluate

Reply:
Added....parameterized in the latest PR update